core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -574,7 +600,8 @@ private[spark] class Executor(
           logInfo(s"Executor interrupted and killed $taskName (TID $taskId), reason: $killReason")
 
           val (accums, accUpdates) = collectAccumulatorsAndResetStatusOnFailure(taskStartTimeNs)
-          val serializedTK = ser.serialize(TaskKilled(killReason, accUpdates, accums))
+          val metricPeaks = WrappedArray.make(metricsPoller.getTaskMetricPeaks(taskId))
+          val serializedTK = ser.serialize(TaskKilled(killReason, accUpdates, accums, metricPeaks))
           execBackend.statusUpdate(taskId, TaskState.KILLED, serializedTK)
 
         case t: Throwable if hasFetchFailure && !Utils.isFatalError(t) =>
```

Comment:
@wypoon Is there any special reason that `metricPeaks` is not send in this case and `CommitDeniedException`?

Reply:
There is no special reason. I saw that in some cases, accumulators are collected and sent in the task result, and in those cases I send the task metric peaks as well; in the other cases, accumulators are not collected so I don't send the task metric peaks either.